components/automate-cli/pkg/verifyserver/models/batchcheck.go

```diff
@@ -78,6 +81,191 @@ type Config struct {
 	APIToken        string      `json:"api_token"`
 }
 
+func (c *Config) appendCertsByIpToNodeCerts(certsByIP *[]config.CertByIP, ipList []string, privateKey, publicKey, adminKey, adminCert string) {
+
+	for _, ip := range ipList {
+		nodeCert := NodeCert{
+			IP:        ip,
+			Key:       privateKey,
+			Cert:      publicKey,
+			AdminKey:  adminKey,
+			AdminCert: adminCert,
+		}
+		c.Certificate.Nodes = append(c.Certificate.Nodes, nodeCert)
+	}
+
+	if certsByIP != nil {
+		for _, certByIP := range *certsByIP {
+			nodeCert := NodeCert{
+				IP:        certByIP.IP,
+				Key:       certByIP.PrivateKey,
+				Cert:      certByIP.PublicKey,
+				AdminKey:  adminKey,
+				AdminCert: adminCert,
+			}
+			c.Certificate.Nodes = append(c.Certificate.Nodes, nodeCert)
+		}
+	}
+}
+
+func (c *Config) PopulateWith(haConfig *config.HaDeployConfig) error {
+	err := haConfig.Verify()
+	if err != nil {
+		return err
+	}
+
+	if haConfig.GetConfigInitials() != nil {
+		configInitials := haConfig.GetConfigInitials()
+		c.SSHUser.Username = configInitials.SSHUser
+		c.SSHUser.PrivateKey = configInitials.SSHKeyFile
+		c.SSHUser.SudoPassword = configInitials.SudoPassword
+		c.Arch = configInitials.Architecture
+		c.Backup.FileSystem.MountLocation = configInitials.BackupMount
+	}
+
+	if haConfig.GetObjectStorageConfig() != nil {
+		objectStorageConfig := haConfig.GetObjectStorageConfig()
+		c.Backup.ObjectStorage.BucketName = objectStorageConfig.BucketName
+		c.Backup.ObjectStorage.AWSRegion = objectStorageConfig.Region
+		c.Backup.ObjectStorage.AccessKey = objectStorageConfig.AccessKey
+		c.Backup.ObjectStorage.SecretKey = objectStorageConfig.SecretKey
+		c.Backup.ObjectStorage.Endpoint = objectStorageConfig.Endpoint
+	}
+
+	c.Hardware.AutomateNodeCount, err = strconv.Atoi(haConfig.Automate.Config.InstanceCount)
+	if err != nil {
+		return err
+	}
+
+	c.Hardware.ChefInfraServerNodeCount, err = strconv.Atoi(haConfig.ChefServer.Config.InstanceCount)
+	if err != nil {
+		return err
+	}
+
+	c.Hardware.PostgresqlNodeCount, err = strconv.Atoi(haConfig.Postgresql.Config.InstanceCount)
+	if err != nil {
+		return err
+	}
+
+	c.Hardware.OpenSearchNodeCount, err = strconv.Atoi(haConfig.Opensearch.Config.InstanceCount)
+	if err != nil {
+		return err
+	}
+
+	c.Certificate.RootCert = haConfig.Automate.Config.RootCA
+	c.Certificate.AutomateFqdn = haConfig.Automate.Config.Fqdn
+	// pre deploy state cs fqdn is same as automate fqdn
+	c.Certificate.ChefServerFqdn = haConfig.Automate.Config.Fqdn
```

Comment:
is there a story for taking from config.toml ChefServerFqdn

Reply:
There is no story, currently.